File: 05_Arrays/src/Plus_One_66.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Plus_One_66 {
    public static void main(String[] args) {
        System.out.println(Arrays.toString(plusOne(new int[] {1,2,3,9})));
        System.out.println(Arrays.toString(plusOne(new int[] {9,9,9,9})));
        System.out.println(Arrays.toString(plusOne(new int[] {1,2,3})));
        System.out.println(Arrays.toString(plusOne(new int[] {4,3,2,2})));
        System.out.println(Arrays.toString(plusOne(new int[] {9})));
    }
    public static int[] plusOne(int[] digits) {
        List<Integer> temp = new ArrayList<>();
        if (digits[digits.length-1] == 9) {
            int k = 1;

            for (int i = 0; i < digits.length; i++) {
                int sum = digits[digits.length-1-i] + k;
                temp.add(sum%10);
                k = sum/10;
            }

            if (k == 1) {
                temp.add(1);
            }
        } else {
            digits[digits.length-1]++;
            return digits;
        }
        int[] ansArr = new int[temp.size()];
        for (int i = 0; i < temp.size(); i++) {
            ansArr[i] = temp.get(temp.size()-1-i);
        }

        return ansArr;

    }
}

File: 05_Arrays/src/Plus_One_66.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
const plusOne = function(digits) {
    if (digits[digits.length-1] < 9) {
        digits[digits.length-1]++;
        return digits;
    } else {
        let k = 1;

        for (let i = 0; i < digits.length; i++) {
            let sum = digits[digits.length-1-i]+k;
            digits[digits.length-1-i] = sum%10;
            k = Math.floor(sum/10);
        }

        if (k) {
            digits.unshift(1);
        }
    }

    return digits;
};

console.log(plusOne([1,2,3]))
console.log(plusOne([1,2,9]))
console.log(plusOne([9,9,9]))
console.log(plusOne([4,3,2,1]))
console.log(plusOne([9]))